assembl/graphql/user.py

```diff
@@ -96,12 +99,34 @@ def mutate(root, args, context, info):
                 if db.query(models.Username).filter_by(
                     username=username
                 ).count():
-                    msg = context.localizer.translate(_(
-                        "We already have a user with this username."))
-                    raise Exception(msg)
+                    raise Exception(u"001: We already have a user with this username.")
 
             user.username_p = username
-            user.real_name_p = args.get('name')
+            name = args.get('name')
+            # only modify the name if it was given in parameter
+            if name is not None:
+                user.real_name_p = name
+
+            old_password = args.get('old_password')
+            new_password = args.get('new_password')
+            new_password2 = args.get('new_password2')
+            # only modify the password if it was given in parameter
+            if old_password is not None and new_password is not None and new_password2 is not None:
+                if not user.check_password(old_password):
+                    raise Exception(u"002: You entered a wrong password.")
```

Comment:
What pattern are you using for the `00N` errors? I'm guessing you use these in the front-end to render the appropriately translated response for errors? If so, you can opt to translate in the backend using the `TranslationFactory` and pass the message along. Furthermore, just during the transition period of moving to graphql, @maparent and I created a sort of "contract" for errors in API format for the V2 interface. I don't know if you wish to expand on that contract or utilize this instead.

The contract is described best in `views/__init__.py` as `JSONError`. You can see multiple uses of the JSONError in `assembl/views/api2/auth.py`. And the consumer of these errors are often in the actions/services in the V2 front-end. This way the front-end just passes the error along to the banner.

Reply:
The pattern is "00N: some text".
We currently still have the issue that the backend translation are using the language selected in the v1 interface (cookie) that can be different from the selected lang in the v2 interface (the two issues we have since several sprints). So you can be in French in the v2 interface but having the errors in English. By doing the translations on the frontend, I don't have the issue. And I think it's best to have the translations of the fields and errors messages in the same po file, so we are sure the translators update all new messages.